File: underhome version java/mysqljava/app/src/main/java/mysql/com/mysqljava/MainActivity.java
package mysql.com.mysqljava;

import android.os.AsyncTask;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import java.sql.*;

public class MainActivity extends AppCompatActivity {
    Connection conn = null;
    TextView b;
    Button r,x, y;
    EditText e;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        b = findViewById(R.id.b);
        r = findViewById(R.id.r);
        e = findViewById(R.id.e);
        x = findViewById(R.id.x);
        y = findViewById(R.id.y);


        r.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Loginn obj = new Loginn();
                obj.execute("");
            }
        });

        x.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                RegistrarUsuario obj = new RegistrarUsuario();
                obj.execute("");
            }
        });

        y.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                EliminarUsuario obj = new EliminarUsuario();
                obj.execute("");
            }
        });

    }

    //para Login


    private class Loginn extends AsyncTask<String,String,String>{
      String msg = "";
      @Override
      protected void onPreExecute(){

      }
      @Override
      protected String doInBackground(String... strings) {
              try {
                  Class.forName("com.mysql.jdbc.Driver");
                  conn = DriverManager.getConnection("jdbc:mysql://ip/bd","usuario_mysql","password_mysql");
                  if(conn == null)
                  {
                      msg="No se pudo Conectar";
                  }
                  else
                  {
                      Statement psd = conn.createStatement();
                      ResultSet rs = psd.executeQuery("select * from a where ass = '" +e.getText().toString()+ "'");
                      if (rs.next()) {
                          b.setText("Bienvenido Usuario!!");
                      }
                      else{
                          b.setText("Usuario no Existe!!");

                      }
                  }
              } catch (Exception e){
                  e.printStackTrace();
              }
              return msg;
          }
          @Override
          protected void onPostExecute(String msg){
          }
          }


          //para Registro de Usuario

    private class RegistrarUsuario extends AsyncTask<String,String,String>{
        String msg = "";
        @Override
        protected void onPreExecute(){

        }
        @Override
        protected String doInBackground(String... strings) {
            try {
                Class.forName("com.mysql.jdbc.Driver");
                conn = DriverManager.getConnection("jdbc:mysql://ip/bd","usuario_mysql","password_mysql");
                if(conn == null)
                {
                    msg="No se pudo Conectar";
                }
                else
                {
                    Statement st = conn.createStatement();
                    st.executeUpdate("insert into a values('" +(e.getText().toString())+ "')");
                    b.setText("registro correcto!!");
                }
            } catch (Exception e){
                e.printStackTrace();
            }
            return msg;
        }
        @Override
        protected void onPostExecute(String msg){
        }
    }

    //para Borrar Usuario

    private class EliminarUsuario extends AsyncTask<String,String,String>{
        String msg = "";
        @Override
        protected void onPreExecute(){

        }
        @Override
        protected String doInBackground(String... strings) {
            try {
                Class.forName("com.mysql.jdbc.Driver");
                conn = DriverManager.getConnection("jdbc:mysql://ip/bd","usuario_mysql","password_mysql");
                if(conn == null)
                {
                    msg="No se pudo Conectar";
                }
                else
                {
                    Statement st = conn.createStatement();
                    st.executeUpdate("delete from a where ass = '" +(e.getText().toString())+ "'");
                    b.setText("Borrado correcto!!");
                }
            } catch (Exception e){
                e.printStackTrace();
            }
            return msg;
        }
        @Override
        protected void onPostExecute(String msg){
        }
    }



}

File: mysqlkotlin/app/src/main/java/conexiones/mysql/com/mysqlkotlin/MainActivity.kt
package conexiones.mysql.com.mysqlkotlin

import android.os.AsyncTask
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*
import java.sql.*
import java.util.*
import android.R.attr.password
import android.os.Handler


class MainActivity : AppCompatActivity() {

    var conn: Connection? = null
    internal var username = "root" // provide the username
    internal var password = "root" // provide the corresponding password

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        //executeMySQLQuery()
        Ass().execute()
        Ass().get()
    }

    fun executeMySQLQuery() {
        var stmt: Statement? = null
        var resultset: ResultSet? = null

        try {
            stmt = conn!!.createStatement()
            resultset = stmt!!.executeQuery("SHOW DATABASES;")

            if (stmt.execute("SHOW DATABASES;")) {
                resultset = stmt.resultSet
            }

            while (resultset!!.next()) {
                println(resultset.getString("Database"))
                a.setText(""+resultset.getString("Database"))
            }
        } catch (ex: SQLException) {
            // handle any errors
            ex.printStackTrace()
        } finally {
            // release resources
            if (resultset != null) {
                try {
                    resultset.close()
                } catch (sqlEx: SQLException) {
                }

                resultset = null
            }

            if (stmt != null) {
                try {
                    stmt.close()
                } catch (sqlEx: SQLException) {
                }

                stmt = null
            }

            if (conn != null) {
                try {
                    conn!!.close()
                } catch (sqlEx: SQLException) {
                }

                conn = null
            }
        }
    }

    fun getConnection() {

            val connectionProps = Properties()
            connectionProps.put("user", username)
            connectionProps.put("password", password)
            try {
                Class.forName("com.mysql.jdbc.Driver").newInstance()

                conn = DriverManager.getConnection(
                        "jdbc:" + "mysql" + "://" +
                                "192.168.20.103" +
                                ":" + "3306" + "/" +
                                "",
                        connectionProps)

                if(conn == null){
                    a.setText("no se conecto")
                }
                else{
                    a.setText(" se conecto")
                }



            } catch (ex: SQLException) {
                // handle any errors
                a.setText(""+ex)

                ex.printStackTrace()
            } catch (ex: Exception) {
                // handle any errors
                a.setText(""+ex)

                ex.printStackTrace()
            }



    }

     inner class Ass: AsyncTask<String, String, String>(){
         var result: String ="";

        override fun onPreExecute() {
            super.onPreExecute()
        }

         override fun doInBackground(vararg params: String?): String {
             val connectionProps = Properties()
             connectionProps.put("user", username)
             connectionProps.put("password", password)
             try {
                 Class.forName("com.mysql.jdbc.Driver").newInstance()

                 conn = DriverManager.getConnection(
                         "jdbc:" + "mysql" + "://" +
                                 "192.168.20.103" +
                                 ":" + "3306" + "/" +
                                 "",
                         connectionProps)

                 if(conn == null){
                     result = "no se conecto"
                 }
                 else{
                     result = " se conecto"
                 }



             } catch (ex: Exception) {
                 // handle any errors

                 result = ex.toString()
                 ex.printStackTrace()
             }
             return result
         }

         override fun onPostExecute(result: String?) {
             super.onPostExecute(result)
             a.setText(""+result)
         }

    }

}

